torch/_inductor/codegen/cpp.py

```diff
@@ -1346,7 +1346,7 @@ def load(self, name: str, index: sympy.Expr):
             else f"{var} + {cexpr_index(index)}"
         )
         loadbuf = "tmpbuf" if non_contiguous else var_expr
-        if is_broadcast:
+        if is_broadcast and dtype not in [torch.bfloat16]:
             if is_mask:
                 loadbuf = f"flag_to_float_scalar({loadbuf})"
             line = f"at::vec::Vectorized<float>(static_cast<float>({loadbuf}))"
```

Comment:
Is it better to enhance this line to incorporate the bfloat16 broadcast instead of adding extra checks in the `else` path?

Reply:
Changed.